codegen/templates/endpoint.tmpl

```diff
@@ -50,21 +52,20 @@ func Handle{{title .Name}}Request(
 	}
 	{{end}}
 
-	headers := map[string]string{}
```

Comment:
What's the advantage to doing that here?

That way we're creating three header objects and parsing/filtering them twice. The headers are inspected and mapped/filtered in the workflow. Seems like we should just pass the http.Header to the workflow and when we want to support non-http endpoints we can create a zanzibar.Header and use that throughout serving.